### DISCUSSION TITLE
ReSync last_activities

### DISCUSSION THREAD

**Discussion by TupeTupeTupeTupe** [jurialmunkey/plugin.video.themoviedb.helper#1273] [Q&A]
I’m experiencing massive stuttering with video playback, and when I checked the log files, I noticed a common factor: every time there’s stuttering, I find the following lines in the log just before the stuttering starts:

info <general>: [plugin.video.themoviedb.helper] ReSync last_activities

What does this do, and is there a setting to disable this syncing? I’m not saying this is necessarily the cause of the stuttering, but I’m trying to rule out all possibilities one by one.

Running kodi 21.1 on Nvidia Shield 2019.

Thanks.

**Comment by jurialmunkey** [comment]
A resync of last_activities occurs when the last_activities cache expires or the cache is forced to refresh. You cannot disable it, it is a necessary part of functioning.

The local last_activities cache is used to compare against the online last_activity timestamps from Trakt to determine if a full a resync of Trakt sync data is required. 

For example, we only want to resync our watched data if we actually watched something. If our local timestamp for watched last_activity and the online timestamp for watched last_activity match then we know our local data is up to date and we don't need to resync (i.e. we last synced after we last watched something).

The last_activity cache expiring just means that we get fresh data from Trakt about the timestamps. It alone is unlikely to cause stuttering as it is a simple request. However, it may also indicate a forced refresh of sync data (e.g. calculating in-progress) which could be more taxing.

Disabling last_activites sync can only make the situation worse because that means that you will unnecessarily resync all data from Trakt every time something requires this data -- until eventually Trakt rate limits or bans your IP because you are syncing large data too frequently.

**Reply by jurialmunkey** [reply]
If you believe there is an issue related to recalculating In-Progress, you can reduce the number of allowable threads it can spawn in the Expert settings section of TMDbHelper.

Reducing number of allowable threads will effectively throttle the process (whereas unlimited just lets it continue to spawn as many threads as your system can take).

If you have very complicated In-Progress/Up-Next lists or lots of shows being tracked on Trakt then this calculation might be a source of stuttering if the threads are uncapped. 

Also note that the option to enable progress indicators syncing from Trakt is particularly taxing. IF you have enabled that I recommend disabling it.

**Reply by TupeTupeTupeTupe** [reply]
Thanks so much for the detailed and thorough response! I’ll give those tweaks you suggested a try and see if they help with my issue. The tricky part here is that the video stutters are unpredictable—they sometimes happen after 5 minutes, sometimes after an hour, and vary from full-on crashes to just a few frame skips. Thanks again for the help; I really appreciate it!
